### DISCUSSION TITLE
QField Version 2.x - It's not possible to use offline basemap on sdcard

### DISCUSSION THREAD

**Discussion by Crapaud04** [opengisch/QField#2878] [Q&A]
It's been a year since I set up an offline visualization solution for vector and raster data over a very large area (1/2 France).

For vectors, it's possible on the tablet (500 mo).
But for rasters how can I do I have 80go orthophoto and 15go scan.
The opening was super fluid and corresponded to my needs and I did the update and everything is broken.

I understand the need to link to file access but for rasters it's only read right ?

Do you have a solution to communicate to me knowing that my tablet fleet is already purchased in 64gb of memory so I cannot transfer my rasters to the internal memory ?

**Comment by nirvn** [comment]
@Crapaud04 , my understanding of your issue is that your device's internal memory where the primary /Android/data/ch.opengis.qfield/files resides doesn't have enough space, is that a proper understanding of your issue?

**Reply by Crapaud04** [reply]
Yes, and I think I'm not the only one using Qfield for this need.
And the weight of the rasters only increasing will be a recurring problem.

**Reply by nirvn** [reply]
When you open QField and tap on the open local file button,  you should see an "Additional files directory" folder:

![Screenshot_20220510-191832](https://user-images.githubusercontent.com/1728657/167626805-695812c2-0815-4eec-a47d-19bb2026c5a8.jpg)

This opens to your external sd card's Android/data/ch.opengis.qfield/ directory. You can copy projects and datasets there and open it in QField.

Are you seeing this folder, and have you tried opening datasets from there?

**Reply by nirvn** [reply]
I might also be misunderstanding you here. Are you referring the datasets that you but in the QField/basemaps folder whereas those are accessible to all projects (ie localized paths)?

**Reply by Crapaud04** [reply]
Yes is basemaps of all projects. 
I tested and i don't acces to this directory in App explorer and winbows explorer same view.  
![image](https://user-images.githubusercontent.com/37582771/167641608-72738c2b-4438-463d-8406-b60b38d014a4.png)

But i click to Additionnal files directory and i see way to the directory in sd card. 

For information : 
My tablet is Samsung Galaxy Tab Active Pro in Android 11.

**Reply by nirvn** [reply]
@Crapaud04 , ok, I'm sure we can find a solution here. 

One thing can be done. You could move the storage from internal to SD card in the app properties, ie:
![Screenshot_20220510-192625_Settings](https://user-images.githubusercontent.com/1728657/167644879-9ab603e9-c781-43c8-a3d2-be458c80081a.jpg)

You find that by long pressing on the QField icon in your launcher and clicking on the (i) button or via settings -> applications -> QField.

As for not seeing any folders in the Android/data directory on windows, have you tried showing hidden files there (I'm not on a windows machine so it's a long shot in the dark :) ).

**Reply by Crapaud04** [reply]
@nirvn I have tested this solution in first before my message.
And i see this directory in internal storage. After change storage location i have don't new directory in my sdcard.

In internal storage, i have the full directory and imported data set my files is in internal storage.

**Reply by nirvn** [reply]
@Crapaud04 , does windows explorer allow you to create a folder in your card's Android/data folder? If so can you try to  create a "ch.opengis.qfield" folder, then a "files" child folder?

**Reply by Crapaud04** [reply]
@nirvn Is not possible in windows and explorer app Android. "access denied" 
After short search on the net this problem is Google and Android 11 ans after.

**Reply by nirvn** [reply]
@Crapaud04 , odd, by most accounts out there you should be able to access content. Is the USB connection set to file transfer (not MTP)? And his your SD card encrypted?

**Reply by Crapaud04** [reply]
@nirvn for information : https://support.google.com/googleplay/android-developer/answer/10467955?hl=fr

I tested to connect on hub usb my sdcard. I see this folder sdcard/Android/data/ch.opengis.qfield/files but this folder is empty. 
So i change storage location yesterday. 

I tested to copy my files in this folder is okey.  And open this files on qfield is okey. 
But my project don't see this files .....

**Reply by nirvn** [reply]
@Crapaud04 , sounds like progress :) I'll get back to you later today with a test build.

**Reply by Crapaud04** [reply]
@nirvn  okey ;)

**Reply by Crapaud04** [reply]
The only problem that strikes me is that the action of change storage does nothing since on the SD card there is nothing.

**Reply by nirvn** [reply]
@Crapaud04 , as far as I can make it, moving the storage to the external drive only moves the app itself, not the files data content.

**Reply by nirvn** [reply]
@Crapaud04 , please test the APKs here https://github.com/opengisch/QField/pull/2886

Upon installation, run the app (QField Beta) once for it to create the directories in your SD card's Android/data/... location. Note that the relevant app directory here will be ch.opengis.qfield_beta on your SD card.

Your basemaps should register just fine 🙂

**Reply by Crapaud04** [reply]
@nirvn  I finich all test on Beta version 2886 . 

I still do not see in the android file explorer nor that of windows the interior of the Sdcard / Android / data folders
So I have to take out my SD card and plug it live on the computer to see these files.
I moved my 2 large files there to the basemaps folder.
I changed in my QGIS mobile project the location folder of the basemaps with notepad++.
example: old path: /storage/9C33-6BBD/qgis/Ortho_50_zs.gpkg
New path: /storage/9C33-6BBD/Android/data/ch.opengis.qfield_beta/files/QField/basemaps/Ortho_50_zs.gpkg

Unless it can do https://docs.qfield.org/fr/how-to/outside-layers/?
If so do you have more screenshots because I couldn't do it.

**Reply by nirvn** [reply]
@Crapaud04 , did you succeed when renaming paths? What is the path where you store your qgs project ?

As you're pointing out there would be better ways to achieve what you want.

**Reply by nirvn** [reply]
For the record, if your ortho dataset is stored in an absolute path, that means it isn't handled as a "basemap" dataset within a localized path (in our case here your sd card/Android/data/.../files/QField/basemaps)

The path for such basemap would be something like localized:ortho.gpkg

**Reply by Crapaud04** [reply]
@nirvn Monday I tested the new version 2.x.
And the absolute path of my rasters didn't work.
While the previous version I had no problem.
It was after that that I asked you.

I just hope I can access this file remotely with my MDM.

**Reply by nirvn** [reply]
@Crapaud04 , any update? Were you successful?

**Reply by Crapaud04** [reply]
@nirvn i update my test tablet is ok but i change my way file in qgs.

But in other modele of adroid tab i acces to sd card directory.

I think samsung lock android sdcard directory.
My problem is partly solved.

For me the solution was to be able to have access to all the sdcard but already it works.
And maybe it will be easier to manage funds with your folder in android/app.
Unless it can do https://docs.qfield.org/fr/how-to/outside-layers/?
If so do you have more screenshots because I couldn't do it.

The delivery of my tablets does not arrive before several months, stock problem.
